src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/histogram.js

```diff
@@ -0,0 +1,683 @@
+        .attr("y", -canvas.margin.left + padding_y)
+        .attr(
+          "x",
+          -canvas.margin.top -
+            canvas.inner.height / 2 +
+            this.textWidth(axis.y.label, fontStyle) / 2
+        )
+        .text(axis.y.label);
+    }
+  }
+
+  /**
+   * Return the text width. Ensures that labels on x axis wont overlap to keeps them readable.
+   */
+  textWidth(text, font) {
+    const canvas = document.createElement("canvas");
+    const context = canvas.getContext("2d");
+    context.font = font;
+    const metrics = context.measureText("  " + text);
+    return metrics.width;
+  }
+
+  /**
+   * Return the extrema of the data and and paddings that ensure data will fit into the
+   * drawing area.
+   *
+   * It traverses through data getting minimal and maximal values, and calculates padding based on
+   * span calculated from above values, multiplied by 10% so that the plot is a little bit smaller
+   * than the container.
+   */
+  extremesAndDeltas() {
+    const dataPoints = this.data();
+    let xMin = dataPoints[0];
+    let xMax = dataPoints[0];
+
+    dataPoints.forEach((value) => {
+      if (value < xMin) {
+        xMin = value;
+      }
+      if (value > xMax) {
+        xMax = value;
+      }
+    });
+
+    const dx = xMax - xMin;
+    const paddingX = 0.1 * dx;
+
+    return { xMin, xMax, paddingX, dx };
+  }
+
+  /**
+   * Return margins for plots drawing area.
+   */
+  margins() {
+    const axis = this.axisSpec();
+    const noXAxis = axis.x.label === undefined;
+    const noYAxis = axis.y.label === undefined;
+
+    if (noXAxis && noYAxis) {
+      return { top: 20, right: 20, bottom: 25, left: 35 };
+    } else if (noYAxis) {
+      return { top: 10, right: 20, bottom: 35, left: 35 };
+    } else if (noXAxis) {
+      return { top: 20, right: 10, bottom: 25, left: 55 };
+    }
+    return { top: 10, right: 10, bottom: 35, left: 55 };
+  }
+
+  /**
+   * Creates HTML div element as container for plot.
+   */
+  createOuterContainerWithStyle(width, height) {
+    const divElem = document.createElementNS(null, "div");
+    divElem.setAttributeNS(null, "class", "vis-histogram");
+    divElem.setAttributeNS(
+      null,
+      "viewBox",
+      0 + " " + 0 + " " + width + " " + height
+    );
+    divElem.setAttributeNS(null, "width", "100%");
+    divElem.setAttributeNS(null, "height", "100%");
+    divElem.setAttributeNS(null, "transform", "matrix(1 0 0 -1 0 0)");
+
+    const addStyleToElem = (attr, stl) => {
+      const style = document.createElement("style");
+      style.innerText = attr + "{" + stl + "}";
+
+      divElem.appendChild(style);
+    };
+
+    const darkStrokeColor = `rgba(255,255,255,0.7)`;
+    const buttonLightColor = `#333`;
+    const darkBtnHoverColor = `rgba(255,255,255,0.5)`;
+    const darkSelectionFill = `#efefef`;
+
+    addStyleToElem(".selection", "rx: 4px;stroke: transparent;");
+    addStyleToElem(
+      "button",
+      `
+            margin-left: 5px; 
```

Comment:
You are using 2-space indentation everywhere, but here we've got 6-space one. Moreover, the closing quote is misaligned. In the next text as well